src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -151,3 +152,172 @@ def update(self):
+        except Exception as e:
+            handle_raw_exception(e)
+
+        otlp_name = self.get_argument_otlp_name()
+        existing_otlps = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+        if existing_otlps is not None:
+            otlp = [p for p in existing_otlps if p["name"].lower() == otlp_name.lower()]
+            if otlp is not None:
+                idx = [i for i, p in enumerate(existing_otlps) if p["name"].lower() == otlp_name.lower()][0]
+                otlp_to_remove = existing_otlps[idx]
+                existing_otlps.remove(otlp_to_remove)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations", value=existing_otlps)
+            else:
+                raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        else:
+            raise ValidationError(f"No otlp entry with --otlp-name {otlp_name} found")
+        
+        self.set_up_enable_open_telemetry_traces()
+        self.set_up_enable_open_telemetry_logs()
+        self.set_up_enable_open_telemetry_metrics()
+        
+    def set_up_otlp(self):
+        if self.get_argument_otlp_name() is not None:
+
+            # Get current containerapp env telemetry properties
+            try:
+                self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+            except Exception as e:
+                handle_raw_exception(e)
+
+            otlp_name = self.get_argument_otlp_name()
+            endpoint = self.get_argument_endpoint()
+            insecure = self.get_argument_insecure()
+            headers = self.get_argument_headers()
+            existing_otlps = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "otlpConfigurations")
+            otlp = {"name": otlp_name} 
+            update = False
+            if existing_otlps is not None:
+                otlp = [p for p in existing_otlps if p["name"].lower() == self.get_argument_otlp_name().lower()]
+                if otlp:
+                    otlp = otlp[0]
+                    update = True
+                else:
+                    otlp = {"name": otlp_name} 
+                    existing_otlps.append(otlp)
+            else:
+                existing_otlps = [otlp]
+            
+            if endpoint:
+                otlp["endpoint"] = endpoint
+            if insecure is not None:
+                otlp["insecure"] = insecure
+
+            if headers is not None:
+                headers_def = []
+                header_pairs = {}
+                for header in headers:
+                    header_value = header.split('=', 1)
+                    header_pairs[header_value[0]] = header_value[1]
```

Comment:
should you check it is valid header format first? the split result may only contains one element, right?

Reply:
we have validator method for validate_otlp_headers, do we need put another round of validation here in this case?